inc/saiswitch.h

```diff
@@ -1827,6 +1893,205 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_NAT_ENABLE,
 
+    /**
+     * @brief SDK/Driver can access device through file system support.
+     *
+     * TRUE - Device access by file system.
+     * FALSE - Device access not supported directly by driver.
+     * Platform adaption device read/write API should be provided by the Host Adapter.
+     * @type bool
+     * @flags READ_ONLY
+     * @validonly SAI_SWITCH_ATTR_TYPE == SAI_SWITCH_TYPE_PHY
+     */
+    SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS,
+
+    /**
+     * @brief Switch hardware access bus MDIO/I2C/CPLD
+     *
+     * @type sai_switch_hardware_access_bus_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @default SAI_SWITCH_HARDWARE_ACCESS_BUS_MDIO
+     * @condition SAI_SWITCH_ATTR_TYPE == SAI_SWITCH_TYPE_PHY and SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS == false
+     */
+    SAI_SWITCH_ATTR_HARDWARE_ACCESS_BUS,
+
+    /**
+     * @brief Platform context information
+     *
+     * Platform context information provided by the host adaptor to driver.
+     * This information is Host adaptor specific, typically used for maintain
+     * synchronization and device information. Driver will give this context back
+     * to adaptor as part of call back sai_switch_register_read/write_fn API.
+     *
+     * @type sai_uint64_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @condition SAI_SWITCH_ATTR_TYPE == SAI_SWITCH_TYPE_PHY and SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS == false
+     */
+    SAI_SWITCH_ATTR_PLATFROM_CONTEXT,
+
+    /**
+     * @brief Platform adaption device read callback function passed to the adaptor.
+     * This is mandatory function for driver when device access not supported by file system.
+     *
+     * Use sai_switch_register_read_fn as read function.
+     *
+     * @type sai_pointer_t sai_switch_register_read_fn
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
```

Comment:
make create_and set?

Reply:
This API and mandatory on create and CREATE_ONLY on mentioned condition

Reply:
what if user want to change this field during runtime ? for example for notification functtion this is possible

Reply:
the function is part of infrastructure, current assumption is that you cannot change it.